### ISSUE TITLE
How to use the pre-trained model on the AudioSet to extract audio features and save them as npy?

### ISSUE THREAD

**Issue by ayameyao** [YuanGongND/ast#98]
Hi, Yuan,

How to use the pre-trained model on the AudioSet to extract audio features and save them as npy?

Thanks,
Guangyao

**Comment by YuanGongND** [comment]
Hi there,

I think the best way is to return `x` at this point. 

https://github.com/YuanGongND/ast/blob/9e3bd9942210680b833b08c39d09f2284ddc4d1d/src/models/ast_models.py#L184

Note - `x` is a sequence, in time-frequency order (i.e., not in time, and not in frequency order), starting with two `[cls]` token. If you don't care sequence, just return after `x = (x[:, 0] + x[:, 1]) / 2`, else return before `x = (x[:, 0] + x[:, 1]) / 2` and do your desired operation.

-Yuan

**Comment by YuanGongND** [comment]
To save, you can just convert `x` to a numpy by `x=x.detach().cpu().numpy()`, and than `np.save(x)`.
